<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1174</width>
    <height>693</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Vertical Puzzle</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>Load a puzzle from file, input a puzzle, or drag symbols from the right:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="InputGrid" name="inputGrid" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_5">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>13</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QWidget" name="solveBar" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>212</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="clearButton">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_7">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="solutionBar" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="solutionBackButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Back</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="solutionPrevButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>&lt;</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>113</width>
             <height>17</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="solutionLabel">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Solution 0 of 0</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>113</width>
             <height>17</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="solutionNextButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
        <zorder>solutionBackButton</zorder>
        <zorder>solutionPrevButton</zorder>
        <zorder>horizontalSpacer</zorder>
        <zorder>solutionLabel</zorder>
        <zorder>horizontalSpacer_2</zorder>
        <zorder>solutionNextButton</zorder>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Minimum</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="Tile" name="label">
          <property name="text">
           <string>A</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="Tile" name="label_2">
          <property name="text">
           <string>B</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="Tile" name="label_3">
          <property name="text">
           <string>C</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="Tile" name="label_10">
          <property name="text">
           <string>D</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="Tile" name="label_4">
          <property name="text">
           <string>E</string>
          </property>
         </widget>
        </item>
        <item row="0" column="5">
         <widget class="Tile" name="label_5">
          <property name="text">
           <string>F</string>
          </property>
         </widget>
        </item>
        <item row="0" column="6">
         <widget class="Tile" name="label_6">
          <property name="text">
           <string>G</string>
          </property>
         </widget>
        </item>
        <item row="0" column="7">
         <widget class="Tile" name="label_9">
          <property name="text">
           <string>H</string>
          </property>
         </widget>
        </item>
        <item row="0" column="8">
         <widget class="Tile" name="label_7">
          <property name="text">
           <string>I</string>
          </property>
         </widget>
        </item>
        <item row="0" column="9">
         <widget class="Tile" name="label_8">
          <property name="text">
           <string>J</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="Tile" name="label_20">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="Tile" name="label_25">
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="Tile" name="label_26">
          <property name="text">
           <string>2</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="Tile" name="label_19">
          <property name="text">
           <string>3</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="Tile" name="label_24">
          <property name="text">
           <string>4</string>
          </property>
         </widget>
        </item>
        <item row="1" column="5">
         <widget class="Tile" name="label_23">
          <property name="text">
           <string>5</string>
          </property>
         </widget>
        </item>
        <item row="1" column="6">
         <widget class="Tile" name="label_21">
          <property name="text">
           <string>6</string>
          </property>
         </widget>
        </item>
        <item row="1" column="7">
         <widget class="Tile" name="label_27">
          <property name="text">
           <string>7</string>
          </property>
         </widget>
        </item>
        <item row="1" column="8">
         <widget class="Tile" name="label_22">
          <property name="text">
           <string>8</string>
          </property>
         </widget>
        </item>
        <item row="1" column="9">
         <widget class="Tile" name="label_28">
          <property name="text">
           <string>9</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="Tile" name="label_11">
          <property name="text">
           <string>+</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="Tile" name="label_12">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="Tile" name="label_13">
          <property name="text">
           <string>*</string>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="Tile" name="label_14">
          <property name="text">
           <string>__</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Maximum</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>17</width>
          <height>13</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>Max number of solutions:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="maxSolutionsLineEdit">
          <property name="text">
           <string>10</string>
          </property>
          <property name="maxLength">
           <number>2</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="verboseCheckBox">
        <property name="text">
         <string>Verbose output</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="solveButton">
          <property name="text">
           <string>Solve</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Console output:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="consoleOutputTextEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
   <zorder>horizontalSpacer_3</zorder>
   <zorder></zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1174</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_puzzle"/>
    <addaction name="actionSave_puzzle"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionLoad_puzzle">
   <property name="text">
    <string>Load puzzle</string>
   </property>
  </action>
  <action name="actionSave_puzzle">
   <property name="text">
    <string>Save puzzle</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Tile</class>
   <extends>QLabel</extends>
   <header>tile.h</header>
  </customwidget>
  <customwidget>
   <class>InputGrid</class>
   <extends>QWidget</extends>
   <header>inputgrid.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
